instrumentation/reactor/reactor-3.1/testing/src/main/groovy/io/opentelemetry/instrumentation/reactor/AbstractReactorCoreTest.groovy

```diff
@@ -1,521 +0,0 @@
-
-    when:
-    (0L..<iterations).forEach { iteration ->
-      def outer = Flux.just("a", "b")
-        .map({ it })
-        .delayElements(Duration.ofMillis(10))
-        .map({ it })
-        .delayElements(Duration.ofMillis(10))
-        .doOnEach({ middleSignal ->
-          if (middleSignal.hasValue()) {
-            def value = middleSignal.get()
-
-            def middle = Flux.just("c", "d")
-              .map({ it })
-              .delayElements(Duration.ofMillis(10))
-              .doOnEach({ innerSignal ->
-                if (innerSignal.hasValue()) {
-                  runWithSpan("inner " + value + innerSignal.get()) {
-                    Span.current().setAttribute("iteration", iteration)
-                  }
-                }
-              })
-
-            runWithSpan("middle " + value) {
-              Span.current().setAttribute("iteration", iteration)
-              middle.subscribe()
-            }
-          }
-        })
-
-      // Context must propagate even if only subscribe is in root span scope
-      runWithSpan("outer") {
-        Span.current().setAttribute("iteration", iteration)
-        outer.subscribe()
-      }
-    }
-
-    then:
-    assertTraces(iterations) {
-      for (int i = 0; i < iterations; i++) {
-        trace(i, 7) {
-          long iteration = -1
-          String middleA = null
-          String middleB = null
-          span(0) {
-            name("outer")
-            iteration = span.getAttributes().get(AttributeKey.longKey("iteration")).toLong()
-            assert remainingIterations.remove(iteration)
-          }
-          span("middle a") {
-            middleA = span.spanId
-            childOf(span(0))
-            assert span.getAttributes().get(AttributeKey.longKey("iteration")) == iteration
-          }
-          span("middle b") {
-            middleB = span.spanId
-            childOf(span(0))
-            assert span.getAttributes().get(AttributeKey.longKey("iteration")) == iteration
-          }
-          span("inner ac") {
-            parentSpanId(middleA)
-            assert span.getAttributes().get(AttributeKey.longKey("iteration")) == iteration
-          }
-          span("inner ad") {
-            parentSpanId(middleA)
-            assert span.getAttributes().get(AttributeKey.longKey("iteration")) == iteration
-          }
-          span("inner bc") {
-            parentSpanId(middleB)
-            assert span.getAttributes().get(AttributeKey.longKey("iteration")) == iteration
-          }
-          span("inner bd") {
-            parentSpanId(middleB)
-            assert span.getAttributes().get(AttributeKey.longKey("iteration")) == iteration
-          }
-        }
-      }
-    }
-
-    assert remainingIterations.isEmpty()
-  }
-
-  def runWithSpan(def publisherSupplier) {
-    runWithSpan("trace-parent") {
-      def tracer = GlobalOpenTelemetry.getTracer("test")
-      def span = tracer.spanBuilder("publisher-parent").startSpan()
```

Comment:
I didn't include the publisher-parent span in the migrated tests since I couldn't reason out any special tested behavior by it, it's just a test-only span. But can add it back if it seems important